Feature/homework 1

VtulkinVladimir opened the pull request:
Разработал консольное приложение. Реализована возможность добавлять автомобили, распечатывать информацию о всех автомобилях, или только об определенных

VtulkinVladimir committed:
Создано приложение

Разработал консольное приложение. Реализована возможность добавлять автомобили, распечатывать информацию о всех автомобилях, или только об определенных

VtulkinVladimir committed:
Автосозданные файлы

не успел пока разобраться с гитИгнором. В будущем исправлю

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,130 @@
+	
+	guard let body = choiseBody() else { return }
+	
+	let year = Int(strongReadLine("Введите год выпуска"))
```
- [ ] год выпуска не обязательное поле, не нужно его строгом просить

VtulkinVladimir replied:
по поводу этого есть вопрос, у меня ведь это поле опциональное, и получается что если у него не получится скастить строчку к инту то он передаст в инициализатор машинки nil. Нужно ли писать отдельную функцию

VtulkinVladimir replied:
отмена вопроса. я в стронгРидЛайне не проверял строчку на пустоту, а действие полезное. сейчас уже очевидно что другая функция нужна))

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,130 @@
+	
+	let year = Int(strongReadLine("Введите год выпуска"))
+	
+	let number = strongReadLine("Введите гос. номер автомобиля")
```
- [ ] и это поле не обязательно

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,130 @@
+	3) Универсал
+	4) Джип
+	5) Хэтчбэк
+	0) Выход
```
зачем при вводе типа кузова есть пункт выход, и почему его тогда нету в запросах других данных?

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,130 @@
+	5) Хэтчбэк
+	0) Выход
+	""") {
+	case "1": return .sedan
```
- [x] выглядит так, что просто в Body нужно добавить инициализатор по порядковому номеру

AlexRaptor reviewed (changes requested):
Поправь замечания в коментах

AlexRaptor commented:
И добавь в README в ветке ФИО

VtulkinVladimir committed:
Работа над ошибками

Добавил инициализатор через индекс. переработал функцию выбора кузова. добавил функцию для чтения необязательных полей

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,138 @@
+func choiseBody() -> Body? {
+	let response = strongReadLine("""
+	Выберите тип кузова из предложенных ниже:
+	1) Седан
```
У тебя есть enum Body, который типизирован стрингом, и значениям которого можно итерироваться. Собери список типов кузова из него, не нужно хардкодить.

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,138 @@
+	print(info)
+}
+
+func sorting() {
```
это не сортировка, а фильтрация

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,138 @@
+	}
+}
+
+func sort(body: Body) {
```
и здесь не сортировка

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,138 @@
+}
+
+func sort(body: Body) {
+	for car in cars {
```
не нужно в цикле, используй конвейер filter{...}.forEach{...}

VtulkinVladimir committed:
Работа над ошибками

Переделал собирание строки с меню, собирание строки с типами кузовов. изменил функцию сортировки

AlexRaptor commented on homework-1/homework-1/main.swift:
```diff
@@ -0,0 +1,151 @@
+
+func filtration() {
+	if let body = choiseBody() {
+		cars.forEach { car in
```
нет, не так... я даже прямо написал в прошлом комментарии, как нужно сделать

> не нужно в цикле, используй конвейер filter{...}.forEach{...}

Смотри, какая тут получается у тебя логика: ты перебираешь все авто и если у текущего авто нужный тип кузова, то ты его распечатываешь.
Это имеет право на жизнь, но я хочу, что бы ты понял, как использовать конвейеры...

- [ ] Нужно, что бы ты реализовал следующую логику: сперва отфильтруй авто, получив список только с нужным типом кузова, а потом уже для каждого выведи его на экран

AlexRaptor commented on homework-1/homework-1/Car.swift:
```diff
@@ -0,0 +1,34 @@
+//
+enum Body: String, CaseIterable
+{
+	case sedan
```
- [ ] добавь тут строковые значения на русском языке

VtulkinVladimir committed:
Работа над ошибками 3

Переделал функцию фильтрации

VtulkinVladimir committed:
Работа над ошибками 4

добавил описание к типам кузова
